build.zig

```diff
@@ -0,0 +1,383 @@
+        lib.defineCMacro("DYNAMIC_POINTER_MASK", "1");
+    }
+
+    if (enable_java_finalization) {
+        lib.defineCMacro("JAVA_FINALIZATION", "1");
+    }
+
+    if (enable_atomic_uncollectable) {
+        lib.defineCMacro("GC_ATOMIC_UNCOLLECTABLE", "1");
+    }
+
+    if (enable_redirect_malloc) {
+        lib.defineCMacro("REDIRECT_MALLOC=GC_malloc", "1");
+        lib.defineCMacro("IGNORE_FREE", "1");
+        if (t.os.tag == .windows) {
+            lib.defineCMacro("REDIRECT_MALLOC_IN_HEADER", "1");
+        } else {
+            lib.defineCMacro("GC_USE_DLOPEN_WRAP", "1");
+        }
+    }
+
+    if (enable_munmap) {
+        lib.defineCMacro("USE_MMAP", "1");
+        lib.defineCMacro("USE_MUNMAP", "1");
+    } else if (enable_mmap) {
+        lib.defineCMacro("USE_MMAP", "1");
+    }
+
+    if (!enable_dynamic_loading) {
+        lib.defineCMacro("IGNORE_DDYNAMIC_LOADING", "1");
+    }
+
+    if (!enable_register_main_static_data) {
+        lib.defineCMacro("GC_DONT_REGISTER_MAIN_STATIC_DATA", "1");
+    }
+
+
+    if (enable_large_config) {
+        lib.defineCMacro("LARGE_CONFIG", "1");
+    }
+
+    if (enable_gc_assertions) {
+        lib.defineCMacro("GC_ASSERTIONS", "1");
+    }
+
+    if (!enable_threads_discovery) {
+        lib.defineCMacro("GC_NO_THREADS_DISCOVERY", "1");
+    }
+
+    if (enable_rwlock) {
+        lib.defineCMacro("USE_RWLOCK", "1");
+    }
+
+    if (enable_checksums) {
+        if (enable_munmap or enable_threads) {
+            @panic("CHECKSUMS not compatible with USE_MUNMAP or threads");
+        }
+        lib.defineCMacro("CHECKSUMS", "1");
+        source_files.appendSlice(&.{
+            "checksums.c"
+        }) catch unreachable;
+    }
+
+    if (build_shared_libs) {
+        source_files.clearAndFree();
+        source_files.appendSlice(&.{
+            "extra/gc.c"
+        }) catch unreachable;
+        if (enable_threads and !(target.isDarwin() or target.isWindows())) {
+            lib.defineCMacro("GC_PTHREAD_START_STANDALONE", "1");
+            source_files.appendSlice(&.{
+                "pthread_start.c",
+            }) catch unreachable;
+        }
+    }
+
+    if (build_shared_libs and !(target.isDarwin() or target.isDragonFlyBSD() or target.isFreeBSD() or target.isNetBSD() or target.isOpenBSD())) {
+    }
+
+    if (build_shared_libs) {
+        lib.defineCMacro("GC_DLL", "1");
+        lib.defineCMacro("GC_NO_VISIBILITY", "1");
+//        lib.defineCMacro("GC_VISIBILITY_HIDDEN_SET", "1");
+//        flags.appendSlice(&.{
+//            "-fvisibility=hidden",
+//        }) catch unreachable;
+    } else {
+        lib.defineCMacro("GC_NOT_DLL", "1");
+    }
+
+    lib.addCSourceFiles(.{
+        .files = source_files.items,
+        .flags = flags.items,
+    });
+    lib.addIncludePath(.{ .path = "include" });
+    lib.linkLibC();
+    if (install_headers) {
+        lib.installHeader("include/gc.h", "gc.h");
+        lib.installHeader("include/gc/cord.h", "gc/cord.h");
```

Comment:
is it possible to make some function for convenience (and reduce possibility of a bug) so that to source specify file name and based destination folder, e.g. installHeaderTo("gc/cord.h", "include") ?

Reply:
Yes, I just added in our own `installHeader` function that does this now so we don't have to repeat the filename in source and destination.

FYI, the first argument is the source, so it's `include/gc.h` in the source git repo. The second argument is the destination and it is implied that it always ends up in the install prefix directory under `include/`. The prefix dirs can be modified with command line args to `zig build`, like so:
```
zig build --help

...

General Options:
  -p, --prefix [path]          Override default install prefix
  --prefix-lib-dir [path]      Override default library directory path
  --prefix-exe-dir [path]      Override default executable directory path
  --prefix-include-dir [path]  Override default include directory path
```